### ISSUE TITLE
_info.cfg cannot be read by filesystem.read_file

### ISSUE THREAD

**Issue by ForestDragon-wesnoth** [wesnoth/wesnoth#6748]
### Game and System Information

- **Version: 1.16.3**
- **Downloaded from: Steam Client**
- **OS: Windows 10**

### Description of the bug

The "filesystem.read_file" lua function (and most likely the other filesystem functions as well) is unable to read the _info.cfg file, as it is likely blacklisted. This makes it significantly harder to make version-checking code for add-ons without hacky workarounds.

### Steps to reproduce the behavior

1. add a filesystem.read_file function (checking either the _info file directly or the folder containing it) to some lua code in any add-on with an _info.cfg file
2. check the contents of the variable to which the function has output the contents
3. _info.cfg and/or its contents should be missing

### Expected behavior

filesystem functions should have read access to _info.cfg, as having it blacklisted only makes things needlessly annoying for UMC creators.

**Comment by CelticMinstrel** [comment]
It seems to me that you're approaching this feature request from the wrong angle.

You want to be able to do a thing and thought of a way to do it that didn't work, so you requested to make that way work, but wouldn't it suffice if some other way was added to do the same thing?

To put it in more concrete terms:

There's no need to de-blacklist info.cfg to satisfy your need, right? All that's needed is an API to return information about installed add-ons. Obviously this would ultimately come from the info.cfg, but the Lua code would be calling something like `wesnoth.get_addon_info()` instead of `filesystem.read_file()`.

**Comment by ForestDragon-wesnoth** [comment]
> It seems to me that you're approaching this feature request from the wrong angle.
> 
> You want to be able to do a thing and thought of a way to do it that didn't work, so you requested to make that way work, but wouldn't it suffice if some other way was added to do the same thing?
> 
> To put it in more concrete terms:
> 
> There's no need to de-blacklist info.cfg to satisfy your need, right? All that's needed is an API to return information about installed add-ons. Obviously this would ultimately come from the info.cfg, but the Lua code would be calling something like `wesnoth.get_addon_info()` instead of `filesystem.read_file()`.
> 
> This also neatly solves any security issues that could potentially arise by just exposing the file. I'm not sure if such things actually exist but I can't be certain that they don't.
Yeah, a API that returns information about installed add-ons would be good enough

**Comment by ForestDragon-wesnoth** [comment]
accidentally closed, my bad
